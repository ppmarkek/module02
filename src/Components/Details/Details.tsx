import { Results } from '../../types';
import '../Search/style.css';

type DetailsProps = {
  detailsInfo?: Results;
  onClickClose?: () => void;
};

export const Details = ({ detailsInfo, onClickClose }: DetailsProps) => {
  return (
    <div className={'RightCart'}>
      <div className={'Cart'}>
        <h3 data-testid="name">Name: {detailsInfo?.name}</h3>
        <p>Created: {detailsInfo?.createdAt}</p>
        <p>Edited: {detailsInfo?.updatedAt}</p>
        <p>Height: {detailsInfo?.height}</p>
        <p>Mass: {detailsInfo?.mass}</p>
      </div>
      <button data-testid="close-icon" id={'close'} onClick={onClickClose}>
        X
      </button>
    </div>
  );
};
