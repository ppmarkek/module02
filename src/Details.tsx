import { DetailsType } from './types';
import './App.css';

type DetailsProps = {
  detailsInfo?: DetailsType;
  onClickClose?: () => void;
};

export const Details = ({ detailsInfo, onClickClose }: DetailsProps) => {
  const handleCloseClick = () => {
    if (onClickClose) {
      onClickClose();
    }
  };

  return (
    <div className={'RightCart'}>
      <div className={'Cart'}>
        <h3>Name: {detailsInfo?.name}</h3>
        <p>Created: {detailsInfo?.createdAt}</p>
        <p>Edited: {detailsInfo?.updatedAt}</p>
        <p>Height: {detailsInfo?.height}</p>
        <p>Mass: {detailsInfo?.mass}</p>
      </div>
      <button id={'close'} onClick={handleCloseClick}>
        X
      </button>
    </div>
  );
};
