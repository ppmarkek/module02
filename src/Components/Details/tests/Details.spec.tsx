import { Details } from '../../Details/Details';
import { fireEvent, render } from '@testing-library/react';
import { resultsMock } from '../../../../fixtures/results';
import React from 'react';

const itemDetails = resultsMock.results[0];
const handleClickClose = jest.fn();

const draw = () => {
  return <Details detailsInfo={itemDetails} onClickClose={handleClickClose} />;
};

describe('Details Component', () => {
  it('renders correctly', () => {
    const component = draw();

    expect(component).toMatchSnapshot();
  });

  it('calls onClickClose when the close button is clicked', () => {
    const { getByTestId } = render(draw());
    const closeIcon = getByTestId('close-icon');
    fireEvent.click(closeIcon);

    expect(handleClickClose).toHaveBeenCalledTimes(1);
  });
});
