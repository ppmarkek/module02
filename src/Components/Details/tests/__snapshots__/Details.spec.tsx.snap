// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component renders correctly 1`] = `
<Details
  detailsInfo={
    {
      "createdAt": "01-01-0101",
      "height": "100",
      "id": "test-id",
      "mass": "100",
      "name": "test name",
      "updatedAt": "02-02-0202",
      "url": "https://url-1.com",
    }
  }
  onClickClose={[MockFunction]}
/>
`;
